File: packages/platform/src/ipc/item.ts
/**
 * 商品管理 IPC 封装 — 列表 / 详情 / 更新。
 *
 * 后端：壳层 `commands/item.rs`（InMemoryItemStore + app::item::ItemService）。
 *
 * @author agent
 * @created 2026-08-13
 */

import { call } from "./invoke";

/** 商品（与 Rust `app::item::Item` 对齐）。 */
export interface Item {
  id: number;
  owner_id: number;
  account_id: string;
  item_id: string;
  title: string;
  price: number;
  desc: string;
  is_polished: boolean;
  is_multi_spec: boolean;
  multi_quantity_delivery: boolean;
  ai_prompt: string;
  has_card: boolean;
  has_default_reply: boolean;
  created_at?: string | null;
}

/** 商品列表查询入参。 */
export interface ItemListRequest {
  owner_id: number;
  page: number;
  page_size: number;
  keyword?: string;
  account_id?: string;
  is_polished?: boolean;
  is_multi_spec?: boolean;
}

/** 查询商品列表（返回 [列表, 总数]）。 */
export function itemList(request: ItemListRequest): Promise<[Item[], number]> {
  return call<[Item[], number]>("item_list", {
    request: {
      owner_id: request.owner_id,
      page: request.page,
      page_size: request.page_size,
      keyword: request.keyword ?? "",
      account_id: request.account_id ?? "",
      is_polished: request.is_polished,
      is_multi_spec: request.is_multi_spec,
    },
  });
}

/** 按商品 ID 查询。 */
export function itemGet(ownerId: number, itemId: string): Promise<Item | null> {
  return call<Item | null>("item_get", { ownerId, itemId });
}

/** 更新商品（AI 提示词）。 */
export function itemUpdate(
  ownerId: number,
  itemId: string,
  aiPrompt?: string,
): Promise<void> {
  return call<void>("item_update", {
    request: { owner_id: ownerId, item_id: itemId, ai_prompt: aiPrompt },
  });
}

/** 商品同步结果（与 Rust `ItemSyncResult` 对齐）。 */
export interface ItemSyncResult {
  synced: number;
  created: number;
  updated: number;
}

/** 从闲鱼平台同步在售商品并入库。 */
export function itemSync(
  ownerId: number,
  accountId?: string,
): Promise<ItemSyncResult> {
  return call<ItemSyncResult>("item_sync", {
    request: {
      owner_id: ownerId,
      account_id: accountId ?? "",
    },
  });
}

/** 闲鱼平台商品详情（与 Rust `PlatformItemDetail` 对齐）。 */
export interface ItemDetail {
  item_id: string;
  title: string;
  desc: string;
  price: number;
  original_price?: number | null;
  images: string[];
  want_count?: number | null;
  browse_count?: number | null;
  item_url: string;
}

/** 从闲鱼平台拉取商品详情。 */
export function itemDetailFetch(ownerId: number, itemId: string, accountId?: string): Promise<ItemDetail> {
  return call<ItemDetail>("item_detail_fetch", {
    request: {
      owner_id: ownerId,
      item_id: itemId,
      account_id: accountId ?? "",
    },
  });
}

File: apps/desktop/src-tauri/src/platforms/xianyu/ipc/item.rs
//! 商品管理 Tauri commands。

use crate::platforms::xianyu::persist::InMemoryItemStore;
use crate::shared::ipc::IpcResponse;
use app::item::{Item, ItemQuery, ItemService};
use common;
use serde::Deserialize;
use std::sync::Arc;
use tauri::State;

pub struct ItemHandle {
    pub store: Arc<InMemoryItemStore>,
}

#[derive(Debug, Deserialize)]
pub struct ItemListRequest {
    pub owner_id: i64,
    pub page: u32,
    pub page_size: u32,
    #[serde(default)]
    pub keyword: String,
    #[serde(default)]
    pub account_id: String,
    pub is_polished: Option<bool>,
    pub is_multi_spec: Option<bool>,
}

#[derive(Debug, Deserialize)]
pub struct ItemUpdateRequest {
    pub owner_id: i64,
    pub item_id: String,
    pub ai_prompt: Option<String>,
}

#[tauri::command]
pub fn item_list(
    state: State<'_, ItemHandle>,
    request: ItemListRequest,
) -> common::OpenDeskResult<IpcResponse<(Vec<Item>, u32)>> {
    let service = ItemService::new(state.store.as_ref());
    let query = ItemQuery {
        page: request.page,
        page_size: request.page_size,
        keyword: request.keyword,
        account_id: request.account_id,
        is_polished: request.is_polished,
        is_multi_spec: request.is_multi_spec,
    };
    let result = service
        .list(request.owner_id, &query)
        .map_err(common::OpenDeskError::wrap)?;
    Ok(IpcResponse::ok(result))
}

#[tauri::command]
pub fn item_get(
    state: State<'_, ItemHandle>,
    owner_id: i64,
    item_id: String,
) -> common::OpenDeskResult<IpcResponse<Option<Item>>> {
    let service = ItemService::new(state.store.as_ref());
    let result = service
        .get(owner_id, &item_id)
        .map_err(common::OpenDeskError::wrap)?;
    Ok(IpcResponse::ok(result))
}

#[tauri::command]
pub fn item_update(
    state: State<'_, ItemHandle>,
    request: ItemUpdateRequest,
) -> common::OpenDeskResult<IpcResponse<()>> {
    let service = ItemService::new(state.store.as_ref());
    service
        .update(request.owner_id, &request.item_id, |item| {
            if let Some(ai_prompt) = &request.ai_prompt {
                item.ai_prompt = ai_prompt.clone();
            }
        })
        .map_err(common::OpenDeskError::wrap)?;
    Ok(IpcResponse::ok(()))
}
